addons/hr_homeworking/wizard/calendar_popover_delete_wizard_homework.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class CalendarPopoverDeleteWizardHomework(models.TransientModel):
+    _name = 'calendar.popover.delete.wizard.homework'
+    _description = 'Calendar Popover Delete Wizard Homework'
+
+
+    record = fields.Many2one('hr.homeworking')
+    date = fields.Date()
+    delete = fields.Selection([('one', 'Delete only this day\'s work location'), ('all', 'Delete this work location for everyweek')], default='one')
+
+    def close(self):
+        if not self.record or not self.delete:
+            pass
+        elif self.delete == 'one':
+            print(self.date)
+            self.record.delete_one_date_default_worklocation(self.date)
+        else:
+            self.record.delete_default_worklocation()
```

Comment:
Make it private + rename into something meaninful (not sure about the suggestion). ``if not self.record or not self.delete`` is not useful now, as the fields should be required. 
```suggestion
    def _remove_default_work_location(self):
        if self.delete == 'one':
            self.record.delete_one_date_default_worklocation(self.date)
        else:
            self.record.delete_default_worklocation()
```

Reply:
If it's private I can't call it from a button :/